### ISSUE TITLE
Better error handling in the processors

### ISSUE THREAD

**Issue by martindekov** [vmware-samples/vcenter-event-broker-appliance#120]
Currently we handle errors coming only from Events coming from vcenter and not the callback functions themselves:
https://github.com/vmware-samples/vcenter-event-broker-appliance/blob/9a9ef34c867dd8e814fd1dafb80a27d0c1d82e85/vmware-event-router/internal/stream/vcenter.go#L88

This can be seen in both processors:
* OpenFaaS:
https://github.com/vmware-samples/vcenter-event-broker-appliance/blob/9a9ef34c867dd8e814fd1dafb80a27d0c1d82e85/vmware-event-router/internal/processor/openfaas.go#L111
* AWS:
https://github.com/vmware-samples/vcenter-event-broker-appliance/blob/9a9ef34c867dd8e814fd1dafb80a27d0c1d82e85/vmware-event-router/internal/processor/aws_event_bridge.go#L176

We should create custom errors by Implementing the Error method:
```go
type ProcessorErr struct {
    processor string
    content string
}
```

Implementing the Error interface:
```go
func (e *ProcessorErr) Error() string { return ....}
```
and the same for OpenFaaS and every processor afterwards. Instead of returning `nil` return the error in the context of: `ProcessorErr.Error("event_bridge","error content")`.

This will enable better integration testing of the event router.

**Comment by embano1** [comment]
Thank you @martindekov ! Can we abstract the error so we don't have a type for every processor? E.g. create a `processor.Error` type in `processor.go` and use across all processors with a constructor `Error.New(processor string, err error)`?

**Comment by martindekov** [comment]
Yeah Michael no problems followed through in the explanation :+1:

**Comment by embano1** [comment]
Closed: #146
